crates/cli-support/src/js/mod.rs

```diff
@@ -1868,24 +1868,28 @@ impl<'a, 'b> SubContext<'a, 'b> {
                                 ),
                             }
                         } else {
-                            let location = if *is_static { "" } else { ".prototype" };
+                            let (location, binding) = if *is_static {
+                                ("", format!(".bind({})", class))
```

Comment:
This runs a mild risk in the future of not being quite as fast as it's creating a new JS function which may bypass optimizations if you hook up a C++ function directly, but I think we can cross that bridge when we get there

Reply:
I know SpiderMonkey (and I think v8 does this too) will understand bound functions and treat them as calls to the original thing rather than inserting any kind of trampoline. My hope is that this means we aren't nullifying and potential host bindings good-ness, but agreed that we can cross that bridge when we get to it.

Reply:
Oh nice! That means this'll work perfectly as is I believe